### PR TITLE
Feature/basic ve ocean

### PR DESCRIPTION
**Pull request by alexcos20**
Add basic veOcean support:
- lock, deposit_for, increase_lock, increase_amount, withdraw (veOcean)
- claim (veFees)
- setAllocate & setBatchAllocate (veAllocation)

Bump ocean-contracts to 1.1.3

### CODE REVIEW

**Comment by alexcos20** [comment]
Integration test are failing, but they are not touched in this PR

```
2022-09-05 19:03:01 ad081f7e6841 ocean_provider.routes.compute[68] ERROR Invalid container values provided on algorithm. Check image, tag and checksum.
```

**Comment by bogdanfazakas** [comment]
> Integration test are failing, but they are not touched in this PR
> 
> ```
> 2022-09-05 19:03:01 ad081f7e6841 ocean_provider.routes.compute[68] ERROR Invalid container values provided on algorithm. Check image, tag and checksum.
> ```

Yep, most likely the checksum we set for the ubuntu container is no longer available on docker hub. Will quickly check

File: src/contracts/ve/veAllocate.ts [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    return allocation
+  }
+
+  /** Get getveAllocation for address, nft, chainId
+   * @param {String} userAddress user address
+   * @param {String} nft NFT address to allocate to
+   * @param {String} chainId chainId of NFT
+   * @return {Promise<number>}
+   */
+  public async getveAllocation(
```
Comment (mihaisc) [inline src/contracts/ve/veAllocate.ts:107]: getVeAllocation

File: src/contracts/ve/veFeeDistributor.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+ * Provides an interface for veOcean contract
+ */
+export class VeFeeDistributor extends SmartContractWithAddress {
+  getDefaultAbi(): AbiItem | AbiItem[] {
+    return veFeeABI.abi as AbiItem[]
+  }
+
+  /**
+   * Claim fees for `userAddress`
+   * Each call to claim look at a maximum of 50 user veOCEAN points.
```
[the remark this answers is not in the document]
Comment (alexcos20) [reply src/contracts/ve/veFeeDistributor.ts]: it's a particular thing on the contracts.  You can only claim up to 50 points in the past, but you might have 100.  at the end, you might end up with having to "claim" twice

File: src/contracts/ve/VeAllocate.ts [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+      this.contract.methods.setAllocation,
+      amount,
+      nft,
+      chainId
+    )
+    return <ReceiptOrEstimate<G>>trxReceipt
+  }
+
+  /**
+   * set a specific percentage of veOcean to a specific nft
```
Comment (mihaisc) [inline src/contracts/ve/VeAllocate.ts:54]: set specific percetage of veOcean to multiple nfts

File: src/contracts/ve/veFeeDistributor.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (mihaisc) [reply src/contracts/ve/veFeeDistributor.ts]: sure but `Each call to claim look at a maximum` , this doesn't sound like English to me

**Review by mihaisc — APPROVED** [review]

**Pull request merged**